zerver/tests/test_user_groups.py

```diff
@@ -2482,10 +2494,46 @@ def check_adding_yourself_to_group(acting_user: str, error_msg: str | None = Non
         do_change_user_group_permission_setting(
             user_group,
             "can_join_group",
-            owners_group,
+            nobody_group,
+            acting_user=None,
+        )
+        self.assertEqual(user_group.can_join_group.named_user_group, nobody_group)
+        check_adding_yourself_to_group("iago", "Insufficient permission")
+
+        do_change_user_group_permission_setting(
+            user_group,
+            "can_add_members_group",
+            admins_group,
+            acting_user=None,
+        )
+        check_adding_yourself_to_group("iago")
```

Comment:
A comment here explaining the case being tested similar to the one in the code block just below would be nice to have

Reply:
Have added the comment above before changing `can_join_group` to `nobody_group`.

Reply:
Looks like you forgot to push; I'm going to merge this and you can add it in a near-term follow-up.

Reply:
I meant there is already a comment for that case.